### DISCUSSION TITLE
Write to file adds null char between each byte

### DISCUSSION THREAD

**Discussion by intelligentandgoodlooking** [alexandrainst/node-red-contrib-ui-upload#18] [Q&A]
I have the **upload node** connected with **join**, and the standard **write file** node, but the resulting file on disk has null characters between each byte. Any ideas what I might have wrong? I have tried using text and binary, also tried changing the write file encoding to ascii, utf8 and utf16. I do not seem to find an option that works nicely. See attached flow, and example files.

![upload](https://github.com/alexandrainst/node-red-contrib-ui-upload/assets/17781311/b5369276-240c-44de-b960-73bb11f4724f)
[Example File.csv](https://github.com/alexandrainst/node-red-contrib-ui-upload/files/12701430/Example.File.csv)
[Example File after upload.csv](https://github.com/alexandrainst/node-red-contrib-ui-upload/files/12701469/Example.File.after.upload.csv)
[flows.zip](https://github.com/alexandrainst/node-red-contrib-ui-upload/files/12701437/flows.zip)

**Comment by Alkarex** [comment]
Hello,
`node-red-contrib-ui-upload` uploads data in chunks. Please try with https://github.com/alexandrainst/node-red-contrib-chunks-to-lines to receive your uploaded data, before sending your data to something else.

There are also [two transfer modes (text and binary)](https://github.com/alexandrainst/node-red-contrib-ui-upload#transfer-types). They are both supported out-of-the-box by `node-red-contrib-chunks-to-lines` but make a difference if you insist on processing the data chunks manually.

See also https://flows.nodered.org/flow/687918dd5cb66a3bfc2a661e15ef4237 for a more comprehensive example
